collect_app/src/main/java/org/odk/collect/android/projects/ProjectCreator.kt

```diff
@@ -0,0 +1,54 @@
+package org.odk.collect.android.projects
+
+import org.json.JSONException
+import org.json.JSONObject
+import org.odk.collect.android.configure.SettingsImporter
+import org.odk.collect.android.preferences.keys.GeneralKeys
+import org.odk.collect.projects.Project
+import org.odk.collect.projects.ProjectsRepository
+import java.net.URL
+
+class ProjectCreator(
+    private val projectImporter: ProjectImporter,
+    private val projectsRepository: ProjectsRepository,
+    private val currentProjectProvider: CurrentProjectProvider,
+    private val settingsImporter: SettingsImporter,
+) {
+
+    fun createNewProject(settingsJson: String): Boolean {
+        val newProject = getNewProject(settingsJson)
+        val savedProject = projectImporter.importNewProject(newProject)
+
+        val settingsImportedSuccessfully = settingsImporter.fromJSON(settingsJson, savedProject.uuid)
```

Comment:
I'm having a little trouble with the seam between `ProjectCreator` and `ProjectImporter`. My reading is:

* `ProjectCreator` is responsible for creating a project based on JSON and setting it up as the current project if needed
* `ProjectImporter` is responsible for setting up all the auxiliary things about a project (directories etc)

Is that right? If that's the case, I feel that `ProjectCreator` should probably perform the `ProjectsRepository#save` and then pass the project along with the JSON settings to the project importer, which should then import the settings from JSON and create directories. I think in that case we'd actually only have a `importProject` method on `ProjectImporter` and `ProjectCreator` could have methods with hardcoded values for creating a demo project and existing project. Now I'm looking at all this, I think the `importExistingProject` code should probably live in a different object (`ExistingProjectImporter` for instance) as that's only really needed once and will go away. 

How does that kind of structure sound? I'll pause my review here, as that might move things around a little.

Reply:
> Is that right?

mostly yeah... `ProjectCreator` does something more than `ProjectImporter`, it's responsible for:
1. Creating new project (project details) based on its url where `ProjectImporter` just takes it as a parameter
2. Calling ProjectImporter.importNewProject() passing the generated project
3. Saving settings for the new project
4. Setting the current project if needed.

I'm not sure which structure we should follow. I would like to leave it as is and get back to it implementing https://github.com/getodk/collect/issues/4579 because then we are going to implement generating details for all cases:
- importing existing project
- importing demo project
- importing projects scanned via QR code
- importing projects added manually

so then there will be no cases where we just save a project to the repository and create dirs (like in those two first cases listed above), we will always need to do pretty the same:
- generate project details
- save project to the repository
- create dirs
- import settings if needed
- sett current project id if needed

so then we I think we will need just one creator/importer do those classes will be merged or one of them removed but this is something I would rather think later as I said working on https://github.com/getodk/collect/issues/4579

Reply:
Hmmm yeah that makes sense. I think I'll end up pulling out the migration stuff and needing to tweak settings import as part of the background jobs (for importing configs with match exactly for instance and migrating existing jobs) so that probably moves closer to this all as well. I'll make sure to rebase on top of this so that we're working in the same direction!